[BE-6679]: Auto re-sign doc updated for API key selection and version… (#1018)
appcircleio/appcircle-docusaurus: docs/publish-module/publish-information/auto-resign-configuration.md

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE6562-updateBuildNumber.png' />

> - **Build Number/Version Code Source**: The defined base build number will be used for versioning during the re-signing process. **App Store**, **TestFlight**, **Google Play** and **Uploaded Binary** are available options

Reviewer: ```suggestion
- **Build Number/Version Code Source**: The defined base build number will be used for versioning during the re-signing process. **App Store**, **TestFlight**, **Google Play**, and **Uploaded Binary** are available options.
```

>     - **App Store**: The build number will be calculated by referencing the latest live version available on the **Apple App Store**.

Reviewer: ```suggestion
    - **App Store**: The build number will be calculated based on the latest live version available on the **Apple App Store**.
```

    - **TestFlight**: The build number will be determined by referencing the latest version available on **TestFlight**.

>     - **Google Play(Android)**: The version code will be set by referencing the latest live version on **Google Play Console**.

Reviewer: ```suggestion
    - **Google Play (Android)**: The version code will be set by referencing the latest live version on **Google Play Console**.
```

>     - **Uploaded Binary**: The build number or version code will be calculated based on the most recently **uploaded binary** to Appcircle.

Reviewer: ```suggestion
    - **Uploaded Binary**: The build number or version code will be calculated from the **most recent binary** uploaded to Appcircle.
```

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE6562-updateVersionNumber.png' />

- **Version Number/Version Name Source**: The defined base version number will be used for versioning during the re-signing process. **App Store**, **TestFlight**, **Google Play** and **Uploaded Binary** are available options
    - **App Store**: The version number will be calculated by referencing the latest live version available on the **Apple App Store**.

>     - **TestFlight**: The version number will be determined by referencing the latest version available on **TestFlight**.

Reviewer: ```suggestion
    - **TestFlight**: The version number will be determined based on the latest version available on **TestFlight**.
```

:::

#### App Store Credential

> Appcircle’s Auto Re-sign feature requires an **App Store Connect** credential. Therefore, selecting a credential is mandatory for both versioning and signing processes. This credential is used to download the necessary signing assets and, if versioning is configured to use App Store data, to retrieve version-related information directly from the store.

Reviewer: ```suggestion
Appcircle’s Auto Re-sign feature requires an **App Store Connect** credential. Therefore, selecting a credential is mandatory for both versioning and signing processes. This credential is used to download the necessary signing assets and retrieve version-related information when versioning is configured to use App Store data.
```
